3-UsingAPIs/Week2/homework/ex4-diceRace.js

```diff
@@ -15,18 +15,30 @@ const rollDie = require('../../helpers/pokerDiceRoller');
 
 function rollDice() {
   const dice = [1, 2, 3, 4, 5];
-  // TODO complete this function; use Promise.race() and rollDie()
+
+  const dicePromises = dice.map((diceNumber) => rollDie(diceNumber));
+
+  return Promise.race(dicePromises);
```

Comment:
You are passing a function to `dice.map()` that takes a number and returns promise returned by `rollDie()`. But `rollDie()` itself also takes a number and returns the same promise. So no need for this intermediate (arrow) function:

```suggestion
  const dicePromises = dice.map(rollDie);

  return Promise.race(dicePromises);
```

You could even shorten it to:

```js
function rollDice() {
  const dice = [1, 2, 3, 4, 5];
  return Promise.race(dice.map(rollDie));
}
```